Assignment_1.py

```diff
@@ -0,0 +1,268 @@
+        return(None)
+
+def bold_mask(bold:np.ndarray,voxel:list = None, r:int = None):
+    
+    """
+    gets the input and produces searchlight mask if applicable.
+    
+    Parameters
+    ----------
+    bold: np.ndarray
+        4-D array of the bold 
+    selected: list
+    
+    r: int
+        
+    Returns
+    -------
+    ret : array_like
+        returns the mask for searchlight
+    """
+
+    if (r is None) and (voxel is None):
+        print('no selection is provided.')
+        mask_cube =np.ones(bold.shape[:3])
+        
+    if (r is None) or (voxel is None):
+        mask_cube =np.ones(bold.shape[:3])
+        warning.warn('Both selection parameters should be defined. \
+\nNo filter is applied')
+
+    else:
+        mask_cube =np.zeros(bold.shape[:3])
+        mask_cube[distmat(mask_cube,voxel) <= r] = 1
+        mask_cube = mask_cube == 1
+    return(mask_cube)
+
+def plot_MRI(image,bold,mask,title=None, ax=None):
+    """
+    gets the input and produces searchlight mask if applicable.
+    
+    Parameters
+    ----------
+    images: int
+    index of the desired image
+    bold: nibabel.nifti1.Nifti1Image
+        BOLD image data
+    mask: array_like
+        Mask that is created to cover unnecessary data
+        
+    Returns
+    -------
+        MRI plots
+    """
+    processed_image = nib.Nifti1Image(image, bold.affine)
+    # Now we can plot it
+    plotting.plot_glass_brain(processed_image,title=title, axes =ax)
+
+# %%
+### THESE ARE PARAMETERS FOR THE MAIN FUNCTION!
+# main function will be created after I understood what should I do :D 
+#VARIABLES:
+
+
+
+
+#%%
+
+# Dir Config
+data_dir = ''
+
+bold_dta_path = 'subj1/bold.nii.gz'
+mask_path = 'subj1/mask4_vt.nii.gz'
+label_path = 'subj1/labels.txt'
+
+## Starting the analysis process:
+
+# data preparation
+bold = nib.load(data_dir + bold_dta_path)
+mask = nib.load(data_dir + mask_path)
+labeled_df = pd.read_csv(data_dir + label_path, sep=' ')
+labeled_df = labeled_df[~labeled_df.labels.isin(['rest','scrambledpix'])]
+chunks = labeled_df.reset_index()\
+    .groupby(['chunks'])\
+    .agg({'index':list}).reset_index()
+#%%
+
+# User inputs:
+x = 20
+y = 30
+z = 10
+r = 10
+
+images = [400,600]
+
+
+selection = create_selection(x=x,y=y,z=z)
+mask_cube = bold_mask(bold = bold.get_fdata(), voxel=selection, r=5)
+bold_array = bold.get_fdata()
+bold_array[~mask_cube] = 0
+
+for chunk in range(len(chunks)):
+    bold_array[:,:,:,chunks.loc[chunk,'index']] = \
+        standardize_data(bold_array[:,:,:,chunks.loc[chunk,'index']])
+
+        
+# %%
+
+
+
+
+
+# RDM analysis process
+
+
+
+bold_array_flatten = bold_array[mask_cube,:]
+
+label_sorting = labeled_df.sort_values('labels').index
+dis_matrix=1-pd.DataFrame(bold_array_flatten[:,label_sorting]).corr('pearson').values
+
+# this two parameters shoudl be obtained from BOLD data
+model_RDM = np.kron((1-np.eye(7)), np.ones((108,108)))
+simi_matrix = np.corrcoef(model_RDM,dis_matrix)[756:,:756]
```

Comment:
you need to flatten the matrices first!

```suggestion
simi_matrix = np.corrcoef(model_RDM.flatten(),dis_matrix.flatten())[1, 1]
```